File: backend/src/main/java/green/shop/diploma/model/Order.java
package green.shop.diploma.model;

import lombok.Data;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Entity
@Data
@Table(name = "order_s")
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    private Date placedAt;

    private String status;

    @ManyToOne
    private Address address;

    @ManyToOne
    private User user;

    @ManyToMany
    private List<Product> products;

    @PrePersist
    void placedAt() {
        this.placedAt = new Date();
    }
}

File: frontend/src/app/models/order.ts
import { Product, User, Address, PricesProduct } from "./";

export class Order {
    id!: number;
    placedAt!: Date;
    status!: string;
    address!: Address;
    products!: Product[];
    _user!: User;

    set user(user: User) {
        this._user = user;
   }

    totalPrice: number;
    pricesProduct: PricesProduct[];
}